google-cloud-bigtable-stats/src/main/java/com/google/cloud/bigtable/stats/BigtableStackdriverExportUtils.java

```diff
@@ -135,13 +183,42 @@ static com.google.monitoring.v3.TimeSeries convertTimeSeries(
         com.google.monitoring.v3.TimeSeries.newBuilder();
     builder.setResource(monitoredResourceBuilder.build());
     builder.setMetric(createMetric(metricName, metricTagKeys, metricTagValues));
-    builder.setMetricKind(createMetricKind(metricType));
-    builder.setValueType(createValueType(metricType));
-    Timestamp startTimeStamp = timeSeries.getStartTimestamp();
-    for (Point point : timeSeries.getPoints()) {
-      builder.addPoints(createPoint(point, startTimeStamp));
+
+    return builder;
+  }
+
+  private static com.google.monitoring.v3.TimeSeries.Builder setupBuilderForGceOrGKEResource(
+      MetricDescriptor metricDescriptor,
+      MonitoredResource gceOrGkeMonitoredResource,
+      TimeSeries timeSeries,
+      String clientId) {
+    List<LabelKey> labelKeys = metricDescriptor.getLabelKeys();
+    String metricName = metricDescriptor.getName();
+    List<LabelKey> metricTagKeys = new ArrayList<>();
+    List<LabelValue> metricTagValues = new ArrayList<>();
+
+    List<LabelValue> labelValues = timeSeries.getLabelValues();
+    for (int i = 0; i < labelValues.size(); i++) {
+      metricTagKeys.add(labelKeys.get(i));
+      metricTagValues.add(labelValues.get(i));
     }
-    return builder.build();
+    metricTagKeys.add(CLIENT_UID_LABEL_KEY);
+    metricTagValues.add(LabelValue.create(clientId));
+
+    com.google.monitoring.v3.TimeSeries.Builder builder =
+        com.google.monitoring.v3.TimeSeries.newBuilder();
+    if (gceOrGkeMonitoredResource == null
```

Comment:
We can move this up to the beginning of this method and do `Precondition.checkNotNull(gceOrGkeMonitoredResource);`